elements/pfe-navigation/src/pfe-navigation.js

```diff
@@ -0,0 +1,322 @@
+import PfeIcon from "../pfe-icon/pfe-icon.js";
+import PfeNavigationItem from "./pfe-navigation-item.js";
+import PfeNavigationMain from "./pfe-navigation-main.js";
+
+if (!Array.prototype.filter){
+  Array.prototype.filter = function(func, thisArg) {
+    'use strict';
+    if ( ! ((typeof func === 'Function' || typeof func === 'function') && this) )
+        throw new TypeError();
+   
+    var len = this.length >>> 0,
+        res = new Array(len), // preallocate array
+        t = this, c = 0, i = -1;
+
+    var kValue;
+    if (thisArg === undefined){
+      while (++i !== len){
+        // checks to see if the key was set
+        if (i in this){
+          kValue = t[i]; // in case t is changed in callback
+          if (func(t[i], i, t)){
+            res[c++] = kValue;
+          }
+        }
+      }
+    }
+    else{
+      while (++i !== len){
+        // checks to see if the key was set
+        if (i in this){
+          kValue = t[i];
+          if (func.call(thisArg, t[i], i, t)){
+            res[c++] = kValue;
+          }
+        }
+      }
+    }
+   
+    res.length = c; // shrink down array to proper size
+    return res;
+  };
+}
+
+class PfeNavigation extends PFElement {
+  static get tag() {
+    return "pfe-navigation";
+  }
+
+  get templateUrl() {
+    return "pfe-navigation.html";
+  }
+
+  get styleUrl() {
+    return "pfe-navigation.scss";
+  }
+
+  get schemaUrl() {
+    return "pfe-navigation.json";
+  }
+
+  get overlay() {
+    return !this._overlay.hasAttribute("hidden");
+  }
+
+  set overlay(state) {
+    if (state) {
+      // Add the overlay to the page
+      this._overlay.removeAttribute("hidden");
+      // This prevents background scroll while nav is open
+      document.body.style.overflow = "hidden";
+    } else {
+      // Remove the overlay from the page
+      this._overlay.setAttribute("hidden", "");
+      // Allow background to scroll again
+      document.body.style.overflow = "auto";
+
+    }
+  }
+
+  constructor() {
+    PFElement._debugLog = true;
+    super(PfeNavigation);
+
+    // Attach functions for use below
+    this._init = this._init.bind(this);
+    this._setVisibility = this._setVisibility.bind(this);
+
+    // -- handlers
+    this._keydownHandler = this._keydownHandler.bind(this);
+    this._resizeHandler = this._resizeHandler.bind(this);
+    this._stickyHandler = this._stickyHandler.bind(this);
+    this._outsideListener = this._outsideListener.bind(this);
+    this._observer = new MutationObserver(this._init);
+
+    // Capture shadow elements
+    this._overlay = this.shadowRoot.querySelector(".pfe-navigation__overlay");
+    this._menuItem = this.shadowRoot.querySelector("pfe-navigation-item");
+
+    // Initialize active navigation item to empty array
+    this._activeNavigationItems = [];
+    this.overlay = false;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    // If this element contains light DOM, initialize it
+    if (this.children.length) {
```

Comment:
The fact that `this._init()` is only called if this condition is met is going to be problematic with Angular. My experience with Angular applications is that when the `connectedCallback` fires in Angular applications, the children are not present so your call to `_init` would never happen in an Angular app.

Look at https://github.com/patternfly/patternfly-elements/blob/master/elements/pfe-tabs/src/pfe-tabs.js#L161 for the approach we've taken on other components.